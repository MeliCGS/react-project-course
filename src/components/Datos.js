
function Datos(nameValue, onResults) {
    const API_KEY = "e0020d8b";
    const inputMovie = nameValue;
    
       fetch(`http://www.omdbapi.com/?apikey=${API_KEY}&s=${inputMovie}`)
        .then(res => res.json())
        .then(results => {
            const { Search } = results
            return onResults(Search)
        })
    return (
        <> 
        </>
    )
}

export default Datos
