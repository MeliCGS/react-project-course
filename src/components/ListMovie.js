import React from 'react'
import Movie from './Movie'

function ListMovie(props) {

    const { movies } = props
    return (
        <div className="MoviesList">
            {
                movies.results.map(movie => {
                    return (
                        <div key={movie.imdbID} className="MoviesList-item">
                            <Movie 
                                id={movie.imdbID}
                                title={movie.Title}
                                poster={movie.Poster}
                                year={movie.Year}
                            ></Movie>
                        </div>
                    )
                })
            }
        </div>
    )
}

export default ListMovie
