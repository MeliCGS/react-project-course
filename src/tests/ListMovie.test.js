import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import ListMovie from '../components/ListMovie'
import Datos from '../components/Datos'

describe('Pruebas en el componente ListMovie', () => {
    test('El componente puede pasar los datos por paramero', done => {
        Datos('avengers', results => {
            const { container } = render(<ListMovie movies={{ results }} />)
            expect(container).toMatchSnapshot()
            done()
        })
    })
})

