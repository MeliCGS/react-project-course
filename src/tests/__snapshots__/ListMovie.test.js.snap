// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ListMovie El componente puede pasar los datos por paramero 1`] = `
<div>
  <div
    class="MoviesList"
  >
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt0848228"
        role="link"
      >
        <h3>
          The Avengers
        </h3>
        <img
          alt="The Avengers"
          src="https://m.media-amazon.com/images/M/MV5BNDYxNjQyMjAtNTdiOS00NGYwLWFmNTAtNThmYjU5ZGI2YTI1XkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SX300.jpg"
        />
        <h2>
          2012
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt4154796"
        role="link"
      >
        <h3>
          Avengers: Endgame
        </h3>
        <img
          alt="Avengers: Endgame"
          src="https://m.media-amazon.com/images/M/MV5BMTc5MDE2ODcwNV5BMl5BanBnXkFtZTgwMzI2NzQ2NzM@._V1_SX300.jpg"
        />
        <h2>
          2019
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt4154756"
        role="link"
      >
        <h3>
          Avengers: Infinity War
        </h3>
        <img
          alt="Avengers: Infinity War"
          src="https://m.media-amazon.com/images/M/MV5BMjMxNjY2MDU1OV5BMl5BanBnXkFtZTgwNzY1MTUwNTM@._V1_SX300.jpg"
        />
        <h2>
          2018
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt2395427"
        role="link"
      >
        <h3>
          Avengers: Age of Ultron
        </h3>
        <img
          alt="Avengers: Age of Ultron"
          src="https://m.media-amazon.com/images/M/MV5BMTM4OGJmNWMtOTM4Ni00NTE3LTg3MDItZmQxYjc4N2JhNmUxXkEyXkFqcGdeQXVyNTgzMDMzMTg@._V1_SX300.jpg"
        />
        <h2>
          2015
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt0118661"
        role="link"
      >
        <h3>
          The Avengers
        </h3>
        <img
          alt="The Avengers"
          src="https://m.media-amazon.com/images/M/MV5BYWE1NTdjOWQtYTQ2Ny00Nzc5LWExYzMtNmRlOThmOTE2N2I4XkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg"
        />
        <h2>
          1998
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt1626038"
        role="link"
      >
        <h3>
          The Avengers: Earth's Mightiest Heroes
        </h3>
        <img
          alt="The Avengers: Earth's Mightiest Heroes"
          src="https://m.media-amazon.com/images/M/MV5BYzA4ZjVhYzctZmI0NC00ZmIxLWFmYTgtOGIxMDYxODhmMGQ2XkEyXkFqcGdeQXVyNjExODE1MDc@._V1_SX300.jpg"
        />
        <h2>
          2010–2012
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt0491703"
        role="link"
      >
        <h3>
          Ultimate Avengers: The Movie
        </h3>
        <img
          alt="Ultimate Avengers: The Movie"
          src="https://m.media-amazon.com/images/M/MV5BMTYyMjk0NTMwMl5BMl5BanBnXkFtZTgwNzY0NjAwNzE@._V1_SX300.jpg"
        />
        <h2>
          2006
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt0803093"
        role="link"
      >
        <h3>
          Ultimate Avengers II
        </h3>
        <img
          alt="Ultimate Avengers II"
          src="https://m.media-amazon.com/images/M/MV5BZjI3MTI5ZTYtZmNmNy00OGZmLTlhNWMtNjZiYmYzNDhlOGRkL2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
        />
        <h2>
          2006
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt0054518"
        role="link"
      >
        <h3>
          The Avengers
        </h3>
        <img
          alt="The Avengers"
          src="https://m.media-amazon.com/images/M/MV5BZWQwZTdjMDUtNTY1YS00MDI0LWFkNjYtZDA4MDdmZjdlMDRlXkEyXkFqcGdeQXVyNjUwNzk3NDc@._V1_SX300.jpg"
        />
        <h2>
          1961–1969
        </h2>
      </a>
    </div>
    <div
      class="MoviesList-item"
    >
      <a
        class="link-movie"
        href="?id=tt2455546"
        role="link"
      >
        <h3>
          Avengers Assemble
        </h3>
        <img
          alt="Avengers Assemble"
          src="https://m.media-amazon.com/images/M/MV5BMTY0NTUyMDQwOV5BMl5BanBnXkFtZTgwNjAwMTA0MDE@._V1_SX300.jpg"
        />
        <h2>
          2012–2019
        </h2>
      </a>
    </div>
  </div>
</div>
`;
